File: app/src/main/java/no/nav/tilbakemeldingsmottak/consumer/email/aad/AADProperties.java
package no.nav.tilbakemeldingsmottak.consumer.email.aad;

import jakarta.validation.constraints.NotEmpty;
import lombok.Data;
import org.springframework.boot.context.properties.ConfigurationProperties;
import org.springframework.validation.annotation.Validated;

@Data
@ConfigurationProperties(prefix = "aad")
@Validated
public class AADProperties {
    @NotEmpty
    private String authority;
    @NotEmpty
    private String clientId;
    @NotEmpty
    private String clientSecret;
    @NotEmpty
    private String email;
    private String tenant;

}

File: app/src/main/kotlin/no/nav/tilbakemeldingsmottak/consumer/email/aad/AADProperties.kt
package no.nav.tilbakemeldingsmottak.consumer.email.aad

import jakarta.validation.constraints.NotEmpty
import org.springframework.boot.context.properties.ConfigurationProperties
import org.springframework.validation.annotation.Validated

@ConfigurationProperties(prefix = "aad")
@Validated
class AADProperties {
    @NotEmpty
    var authority: String = ""

    @NotEmpty
    var clientId: String = ""

    @NotEmpty
    var clientSecret: String = ""

    @NotEmpty
    var email: String = ""

    var tenant: String? = null
}
